### PR TITLE
REF Code cleanup / update

### PR DESCRIPTION
**Pull request by mattwire**
- Use Civi classloader and move files to Civi/FlexMailer to match namespace
- Regenerate civix
- Switch to generic settings form
- Cleanup flexmailer.php

### CODE REVIEW

File: flexmailer.php [hunk 1]
```diff
@@ -143,13 +143,13 @@ function flexmailer_civicrm_alterSettingsFolders(&$metaDataFolders = NULL) {
...
-    'label' => ts('Flexmailer Settings', array('domain' => 'org.civicrm.flexmailer')),
+    'label' => E::ts('Flexmailer Settings'),
     'name' => 'flexmailer_settings',
     'permission' => 'administer CiviCRM',
-    'child' => array(),
+    'child' => [],
     'operator' => 'AND',
     'separator' => 0,
-    'url' => CRM_Utils_System::url('civicrm/admin/flexmailer', 'reset=1', TRUE),
+    'url' => CRM_Utils_System::url('civicrm/admin/setting/flexmailer', 'reset=1', TRUE),
```
Comment (totten) [inline flexmailer.php:137]: Just to verbalize: I think this checks-out. Ordinarily, one might be concerned about changing the path for a page. However, in this case, the item is registered via `hook_civicrm_navigationMenu` which (IIRC) means that the menu should always be up-to-date with the declaration therein.

By contrast, some extensions use the installer to insert a row into `civicrm_navigation` - with that technique, would be more concerned about downstream folks fiddling with the configuration data. But given the hook that's actually used here, it doesn't seem like a concern

File: info.xml [hunk 2]
```diff
@@ -23,8 +23,11 @@
     to provide richer email features.
   </comments>
   <compatibility>
-    <ver>5.13</ver>
+    <ver>5.19</ver>
   </compatibility>
+  <classloader>
+    <psr4 prefix="Civi\" path="Civi"/>
+  </classloader>
```
Comment (totten) [inline info.xml:30]: Huzza for -27 SLOC on `_flexmailer_autoload()`!

To replace it, did you try just:

```xml
 <psr4 prefix="Civi\FlexMailer\" path="src"/>
```

Then it's not necessary to rename the files. The benefit of PSR-4 (versus the older PSR-0) is that it allows cleaner/less-redundant file-trees.

File: templates/CRM/Flexmailer/Form/FlexmailerAdmin.tpl [hunk 3]
```diff
@@ -1,20 +0,0 @@
...
-          <span class="description">
-            {ts}For greater backward-compatibility, process "<code>traditional</code>" mailings with the CiviMail's hard-coded BAO.{/ts}<br/>
-            {ts}For greater forward-compatibility, process "<code>traditional</code>" mailings with Flexmailer's extensible pipeline.{/ts}<br/>
-          </span>
-        </td>
-      </tr>
-    </table>
-    <div class="crm-submit-buttons">{include file="CRM/common/formButtons.tpl" location="bottom"}</div>
-  </div>
-{/crmScope}
```
Comment (totten) [inline templates/CRM/Flexmailer/Form/FlexmailerAdmin.tpl:20]: Woot -40 SLOC on `FlexmailerAdmin.{php,tpl}`

File: info.xml (hunk 2 above)
Comment (mattwire) [reply info.xml:30]: @totten This took a while for me to circle back around to! But you are right it works so I've updated the PR

**Comment by mattwire** [comment]
Hey @totten This should be good for merge now I think? I've added the psr4 prefix so it's a much simpler PR now.

**Comment by mlutfy** [comment]
I did a code review and looks good.

+1 on using the settings and standardizing the path. It's a change in the path, but it's not something we access often, and it has a navigation menu item.

**Comment by seamuslee001** [comment]
Changes are fine with me I don't have permission to merge but would if I could

**Pull request merged**